packages/blockchain-wallet-v4-frontend/src/modals/Swap/SuccessfulSwap/index.tsx

```diff
@@ -18,46 +20,156 @@ const Wrapper = styled(FlyoutWrapper)`
   flex-direction: column;
 `
 
+const StyledText = styled(Text)`
+  margin-top: 24px;
+`
+
+const StyledSubText = styled(Text)`
+  margin-top: 4px;
+`
+
+const StyledEarnSubText = styled(Text)`
+  margin-top: 30px;
+  padding: 0 50px;
+  text-align: center;
+`
+
+const StyledDoneButton = styled(Button)`
+  margin-top: 16px;
+`
+
+const StyledEarnButton = styled(Button)`
+  margin-top: 24px;
+`
+
 const SuccessfulSwap: React.FC<Props> = (props) => {
-  if (!props.order) return null
+  const {
+    analyticsActions,
+    close,
+    handleClose,
+    interestEligible,
+    interestRate,
+    isRewardsFlowAfterSwapEnabled,
+    order
+  } = props
+  if (!order) return null
+
+  const swappedCurrency = getOutput(order)
```

Comment:
performance boost if you wrap this and the `const`s below it in a `useMemo()`

Reply:
Sure, thanks.